monitoring/v1/monitoring.proto

```diff
@@ -51,6 +51,71 @@ service MonitoringService {
             body: "*"
         };
     }
+
+    // Get the memory usage metrics for the deployment based on the given request.
+    // Required permissions:
+    // - monitoring.metrics.get on the deployment identified by the given deployment ID.
+    rpc GetDeploymentMemoryUsageMetrics(GetDeploymentMetricsRequest) returns (DeploymentMetrics) {
+        option (google.api.http) = {
+            get: "/api/monitoring/v1/metrics/memory"
+        };
+    }
+
+    // Get the CPU usage metrics for the deployment based on the given request.
+    // Required permissions:
+    // - monitoring.metrics.get on the deployment identified by the given deployment ID.
+    rpc GetDeploymentCPUUsageMetrics(GetDeploymentMetricsRequest) returns (DeploymentMetrics) {
+        option (google.api.http) = {
+            get: "/api/monitoring/v1/metrics/cpu"
+        };
+    }
+
+    // Get the disk usage metrics for the deployment based on the given request.
+    // Note: The response will contain an empty list for coordinators.
+    // Required permissions:
+    // - monitoring.metrics.get on the deployment identified by the given deployment ID.
+    rpc GetDeploymentDiskUsageMetrics(GetDeploymentMetricsRequest) returns (DeploymentMetrics) {
+        option (google.api.http) = {
+            get: "/api/monitoring/v1/metrics/disk"
+        };
+    }
```

Comment:
Why not combining into 1  GetDeploymentMetrics(...)
and add the metrics type as a request parameter?

Reply:
If we add 2 fields we can make it even easier imo, and the result doesn't contain Coordinator/DBServer anymore, but the dashboard need to make 2 calls:
- MetricsType [CPU/Memory/Disk]
- ServerType [Coordinator/DBServer]

Reply:
Done, updated